controllers/grid/users/reviewer/form/AdvancedSearchReviewerForm.php

```diff
@@ -256,6 +283,48 @@ public function fetch($request, $template = null, $display = false)
         return parent::fetch($request, $template, $display);
     }
 
+    /**
+     * @copydoc Form::execute()
+     */
+    public function execute(...$functionArgs)
+    {
+        /** @var \PKP\submission\reviewAssignment\ReviewAssignment $reviewAssignment */
+        $reviewAssignment = parent::execute(...$functionArgs);
+        $request = Application::get()->getRequest();
+
+        if (!$request->getContext()->getData('reviewerSuggestionEnabled')) {
+            return $reviewAssignment;
+        }
+
+        $reviewerId = $reviewAssignment->getData('reviewerId');
+
+        if ($this->reviewerSuggestion?->existingReviewerRole
+            && $this->reviewerSuggestion->existingUser->getId() == $reviewerId) {
+
+            $this->reviewerSuggestion->markAsApprove(
+                Carbon::now(),
+                $reviewerId,
+                $request->getUser()->getId()
+            );
+
+            return $reviewAssignment;
+        }
+
+        // TODO :   This check probably needed to port to `CreateReviewerForm` and `EnrollExistingReviewerForm` also
+        //          as this case can be duplicated from there also . 
+        //          Probably better to refactor to a trait or repository class ?
+        $suggestion = ReviewerSuggestion::query()
```

Comment:
This checks whether there is a reviewer suggestion with the same email as the selected existing Reviewer that the user assigns to the submission. 

Do we need to consider the un-assignment of a reviewer, so that is this is the case, the suggestion should be, again, displayed in the "Suggested Reviewers" panel?

Reply:
> Do we need to consider the un-assignment of a reviewer, so that is this is the case, the suggestion should be, again, displayed in the "Suggested Reviewers" panel?

I dont think we need for this check . As once a suggestion get approved, it will be marked as approves. That will lead to user creation, adding reviewer role and other actions . Now if that reviewer got unassigned, even so the suggestion will be still marked as approved as a reviewer link has been established and marked as approved so it will be visible to select again as we will only pull the list of unapproved suggestions .